app.js

```diff
@@ -0,0 +1,780 @@
+  displayDamage(10);
+  return 10;
+}
+
+function fortify() {
+  if(players["player"].shield >= 5) {
+    setPlayerShield(players["player"].shield + 5);
+  }
+}
+
+//---------------------------------------------------------------------------------
+
+function bossTurn() {
+  isPlayerTurn = false;
+  if(checkLifePoints()) {
+    return;
+  }
+  setBossShield(0);
+  //pick and execute an attack
+  var bossEvent = bossAttacks[randomNumber(0,bossAttacks.length-1)];
+  setTimeout(function() {
+    setProperty("boss_description","text","Time Keeper is using " + bossEvent.name);
+  }, 1000);
+  setTimeout(function() {
+    window[bossEvent.name]();
+  },3000);
+  setTimeout(function() {
+    setProperty("boss_description","text", "");
+  }, 4000);
+  setTimeout(function() {
+    useStatus();
+    startTurn();
+  }, 5000);
+  setTimeout(function() {
+    
+  }, 5000);
+}
+
+//Boss cards
+function strike() {
+  setPlayerHealth(players["player"].health - 15);
+  displayDamage(15);
+}
+function slash() {
+  players["player"].status.push(JSON.parse(JSON.stringify(statusEffects.fire)));
+  setPlayerHealth(players["player"].health-10);
+  showStatus();
+  displayDamage(10);
+}
+
+function reinforce() {
+  setBossShield(players["boss"].shield + 10);
+  setPlayerHealth(players["player"].health - 10);
+  displayDamage(10);
+}
+
+//-----------------------------------------------------------------------------
+
+
+
+//Deck creation
+
+//data split for each "screen"
+var showCase = [
+  [cards.spirit,cards.club,cards.fortify,cards.cleanse,cards.blankCard], 
+  [cards.smite, cards.block, cards.draw,cards.heal,cards.flame], 
+  [cards.smear,cards.dash,cards.chain,cards.fortune,cards.grow]
+];
+
+//starting idx
+var showCaseIdx = 1;
+
+
+function switchShowCase(button) {
+  if (button == "right") {
+    if ((showCaseIdx + 1) > 2) {
+      showCaseIdx = 0;
+    }
+    else {
+      showCaseIdx++;
+    }
+  }
+  if (button == "left") {
+    if((showCaseIdx - 1) < 0) {
+      showCaseIdx = 2;
+    }
+    else {
+      showCaseIdx--;
+    }
+  }
+  for(var i=0; i<5; i++) {
+    setProperty("sample_"+i,"image", showCase[showCaseIdx][i].image);
+  }
+}
+
+function checkCount() {
+  if (deck.length == 15) {
+    shuffle(deck);
+    setScreen("fight_screen");
+    startTurn();
+    float();
+  }
+}
+
+
+//button events
+onEvent("right_button","click",function() {
+  switchShowCase("right");
+});
+onEvent("left_button","click",function() {
+  switchShowCase("left");
+});
+
+function addCardToDeck(cardIdx) {
+  if(showCase[showCaseIdx][cardIdx].name != "none") {
+    deck.push(showCase[showCaseIdx][cardIdx]);
+    checkCount();
+    setProperty("deck_count", "text", "Count:    "+deck.length + "/15");  
+  }
+}
+//events for each sample card
+onEvent("undo_button","click",function() {
+  deck.pop();
+  setProperty("deck_count", "text", "count:    "+deck.length + "/15");
+});
+onEvent("sample_0","click",function() {
+  addCardToDeck(0)
+});
```

Comment:
So like I said above I think you want to load a `card` asap and start using it in the code instead of the big `showCase` card lookup.

I think you can continue to make these calls pretty generic though

```
onEvent("sample_0","click",function() {
  addCardToDeck(lookupCard(0));
});

function lookupCard(idx) {
  return showCase[showCaseIdx][cardIdx];
} 
```

Reply:
Like I said above, `sample` ids should change to `showcase` now